### PR TITLE
Integration smoke test

### PR DESCRIPTION
**Pull request by whscullin**
Adds a simple integration test - it loads a minimal Malloy file, runs the file, and retrieves the result.

Part of this was to add some plumbing to the `malloy.runQueryFile` so that it waits for the results and returns them as result of the command.

Most of the work was getting cloud build image that allowed vscode to start up, although I squashed a lot of flailing.

### CODE REVIEW

File: integration/tests/extension.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import * as assert from 'assert';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+
+import {ResultJSON} from '@malloydata/malloy';
+
+describe('Smoke tests', function () {
+  this.timeout(20000);
+
```
Comment (narreola) [inline integration/tests/extension.test.ts:33]: Is this needed to make sure the VS code is up and running? Is there another way different than a fixed timeout to achieve this?

File: integration/tests/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import * as path from 'path';
+import * as Mocha from 'mocha';
```
Comment (narreola) [inline integration/tests/index.ts:25]: (just an idea, not a PR blocker) would it be worth it at some point to use Jest for this too? I was playing with runCLI a bit.

Comment (narreola) [reply integration/tests/index.ts:25]: its not super important but I think it would be nice to use the same framework across all tests.

File: scripts/build_cloud_build_image.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+pushd cloudbuild/builder
+
+# Get latest VS Code package
+wget 'https://code.visualstudio.com/sha/download?build=stable&os=linux-deb-x64' -O vscode.deb
```
Comment (narreola) [inline scripts/build_cloud_build_image.sh:6]: I'm not very familiar with cloud build, but could we have more images and run in multiple different environments (covering multiple OS's and architectures) this may be a test we run asynchronously.

File: src/extension/commands/run_query_file.ts [hunk 4]
```diff
@@ -24,14 +24,15 @@
...
 import {runMalloyQuery} from './run_query_utils';
 import {WorkerConnection} from '../worker_connection';
+import {ResultJSON} from '@malloydata/malloy';
 
-export function runQueryFileCommand(
+export async function runQueryFileCommand(
   worker: WorkerConnection,
   queryIndex = -1
-): void {
+): Promise<ResultJSON | undefined> {
```
Comment (narreola) [inline src/extension/commands/run_query_file.ts:32]: Just curious, why is this change needed?

**Review by narreola — APPROVED** [review]
overall LGTM

File: integration/tests/extension.test.ts (hunk 1 above)
Comment (whscullin) [reply integration/tests/extension.test.ts:33]: Not entirely, I just set it much larger for the purposes of not having false negative because a query took some time. Not really an issue with duckdb, it could be probably be dialed back.

File: integration/tests/index.ts (hunk 2 above)
Comment (whscullin) [reply integration/tests/index.ts:25]: Yeah, I read a post about how to use jest, I don't love using undocumented APIs but it's probably a minimal risk. I did switch from from the TDD UI to BDD UI so that we would be using `describe()` and `it()` instead of `suite()` and `test()` at least.

File: scripts/build_cloud_build_image.sh (hunk 3 above)
Comment (whscullin) [reply scripts/build_cloud_build_image.sh:6]: We can have a variety of images, but running Windows or macOS for testing would not be easy given the Docker on Linux VM underpinnings of Cloud Build.

File: src/extension/commands/run_query_file.ts (hunk 4 above)
Comment (whscullin) [reply src/extension/commands/run_query_file.ts:32]: VS Code will plumb results back from commands, but we were not returning anything. I use the to validate the results of the query that was run. I updated all the query running commands to return results, although I'm only using the one at the moment.
